# Graded assignment 1
Mirte Hanssen U526393, 2008317


# Packages

We need the package `tidyverse`

In [15]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [16]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3          V4        V5        V6        V7       
1 0.2318017 0.1377585 0.073982906 0.1891970 0.5952601 0.5024419 0.3709069
2 0.8558872 0.4933178 0.009001817 0.4662642 0.5898956 0.8247336 0.5303931
  V8         V9         V10        V11       V12       V13       V14      
1 0.01212862 0.02553835 0.08762297 0.2665659 0.4963226 0.4517630 0.7652984
2 0.51642357 0.93950623 0.15558162 0.9678820 0.7496860 0.7052398 0.9868000
  V15       V16        V17       V18       V19       V20      
1 0.5682117 0.48605127 0.5458712 0.1869023 0.8235961 0.6878881
2 0.4365162 0.02947346 0.3528397 0.2527104 0.7733575 0.1210212

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [28]:
df2 <- df1 %>%
  gather(Key,Value,V1:V20)

head(df2)

Key Value      
1 V1  0.231801680
2 V1  0.855887166
3 V2  0.137758498
4 V2  0.493317754
5 V3  0.073982906
6 V3  0.009001817

## Assignment 2

We will generate some data:

In [18]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [19]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [27]:
df5 <- full_join(df3,df4,by= "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook